From c14554f4afaadf6e551026297c1905d9aa3ae3fd Mon Sep 17 00:00:00 2001
From: Chris Robinson <chris.kcat@gmail.com>
Date: Thu, 13 Feb 2025 19:57:01 -0800
Subject: [PATCH] Check __wargv for null before accessing it

---
 core/helpers.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/helpers.cpp b/core/helpers.cpp
index cff864c5e7..96fca1065e 100644
--- a/core/helpers.cpp
+++ b/core/helpers.cpp
@@ -102,6 +102,12 @@ const PathNamePair &GetProcBinary()
 
         fullpath.resize(len);
 #else
+        if(__argc < 1 || !__wargv)
+        {
+            ERR("Failed to get process name: __argc = {}, __wargv = {}", __argc,
+                static_cast<void*>(__wargv));
+            return PathNamePair{};
+        }
         const WCHAR *exePath{__wargv[0]};
         if(!exePath)
         {
